Pollo/Supporting/BannerController.swift

```diff
@@ -7,15 +7,177 @@
+                                    } else {
+                                        let alertController = self.createAlert(title: "Error", message: "Failed to delete session. Try again!")
+                                        self.present(alertController, animated: true, completion: nil)
+                                    }
+                                case .error(let error):
+                                    print(error)
+                                    let alertController = self.createAlert(title: "Error", message: "Failed to delete session. Try again!")
+                                    self.present(alertController, animated: true, completion: nil)
+                                }
+                            }
+                        }
+                    } else {
+                        self.delegate?.pollsDateViewControllerWasPopped(for: self.userRole)
+                    }
+        }
+
+
+        // ok
+
+
+    }
+
+    func getMembers(with id: String) -> Future<Response<[GetMemberResponse]>> {
+        return networking(Endpoint.getMembers(with: id)).decode()
+    }
+
+    func deleteSession(with id: String) -> Future<DeleteResponse> {
+        return networking(Endpoint.deleteSession(with: id)).decode()
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override var preferredStatusBarStyle: UIStatusBarStyle {
+        return .lightContent
+    }
+}
+
+extension PollingViewController: NavigationTitleViewDelegate {
+    func navigationTitleViewNavigationButtonTapped() {
+        guard userRole == .admin else { return }
+        let pollsDateAttendanceArray = pollsDateArray.map { PollsDateAttendanceModel(model: $0, isSelected: false) }
+        getMembers(with: session?.id ?? "").observe { [weak self] result in
+            guard let `self` = self else { return }
+            DispatchQueue.main.async {
+                switch result {
+                case .value(let response):
+                    let groupControlsVC = GroupControlsViewController(session: self.session, pollsDateAttendanceArray: pollsDateAttendanceArray, numMembers: response.data.count, delegate: self)
+                    self.navigationController?.pushViewController(groupControlsVC, animated: true)
+                case .error(let error):
+                    print(error)
+                    let alertController = self.createAlert(title: "Error", message: "Failed to load data. Try again!")
+                    self.present(alertController, animated: true, completion: nil)
+                }
+            }
+        }
+    }
+}
+
+// MARK: - GroupControlsViewControllerDelegate
+extension PollingViewController: GroupControlsViewControllerDelegate {
+    func groupControlsViewControllerDidUpdateSession(_ session: Session) {
+        self.session = session
+    }
+
+
+}
+
 public class BannerController {
 
     static let shared = BannerController()
```

Comment:
lets also make a `private init() {}` so that others can't initialize this and only used `BannerController.shared`